frontend/hub/api/utils.tsx

```diff
@@ -7,11 +7,49 @@ import {
   postHubRequest,
   putHubRequest,
 } from './request';
+import { useTranslation } from 'react-i18next';
+import { requestGet } from '../../common/crud/Data';
 
 function getBaseAPIPath() {
   return process.env.HUB_API_PREFIX;
 }
 
+export function useRepositoryBasePath(name: string, pulp_href?: string): Promise<unknown> {
+  const { t } = useTranslation();
+  return Promise.all([
+    pulp_href
+      ? Promise.resolve({ name, pulp_href })
+      : requestGet<{ data: { results: unknown } }>(
+          pulpAPI`/repositories/ansible/ansible/?name=${name}&limit=1`
+        ).then(firstResult),
+    requestGet<{ data: { results: unknown } }>(
+      pulpAPI`/distributions/ansible/ansible/?name=${name}&limit=1`
+    ).then(firstResult),
+  ]).then(async ([repository, distribution]) => {
+    if (!repository) {
+      return Promise.reject(t`Failed to find repository ${name}`);
+    }
+
+    if (distribution && distribution.repository === repository.pulp_href) {
+      return distribution.base_path;
+    }
+
+    distribution = await requestGet(
+      pulpAPI`/distributions/ansible/ansible/?repository=${repository.pulp_href}&ordering=pulp_created&limit=1`
+    ).then(firstResult);
+
+    if (!distribution) {
+      return Promise.reject(t`Failed to find a distribution for repository ${name}`);
+    }
+
+    return distribution.base_path;
+  });
+}
+
+function firstResult({ data: { results } }) {
```

Comment:
yeaaah, if you want to type this right, this will have to be 

```suggestion
function firstResult<T>({ data: { results: T[] } }) {
```

and then instead of calling `.then(firstResult)` it'd be `.then(firstResult<AnsibleRepository>)` (and the type is different depending on distro/repo).

Or you could try 

```suggestion
function firstResult({ data: { results } }: { data: { results: unknown[] } }) {
```

Reply:
(might also be a good idea to take all those `{ data: { results: unknown } }` and `{ data: { results: unknown[] } }` and move the definition up into a `type Results = { data: { results: unknown[] } };` and use that instead)